Packs/Code42/Integrations/Code42/Code42.py

```diff
@@ -15,177 +18,189 @@
-SECURITY_ALERT_HEADERS = ['Type', 'Occurred', 'Username', 'Name', 'Description', 'State', 'ID']
+SECURITY_EVENT_HEADERS = [
+    "EventType",
+    "FileName",
+    "FileSize",
+    "FileHostname",
+    "FileOwner",
+    "FileCategory",
+    "DeviceUsername",
+]
+SECURITY_ALERT_HEADERS = ["Type", "Occurred", "Username", "Name", "Description", "State", "ID"]
 
 
 class Code42Client(BaseClient):
     """
-    Client will implement the service API, should not contain Demisto logic.
+    Client will implement the service API, should not contain Cortex XSOAR logic.
     Should do requests and return data
     """
 
     def __init__(self, sdk, base_url, auth, verify=True, proxy=False):
         super().__init__(base_url, verify=verify, proxy=proxy)
-        # Create the Code42 SDK instnace
-        self._sdk = sdk.create_using_local_account(base_url, auth[0], auth[1])
-        py42.settings.set_user_agent_suffix("Demisto")
+        # Create the Code42 SDK instance
+        self._sdk = sdk or py42.sdk.from_local_account(base_url, auth[0], auth[1])
+        py42.settings.set_user_agent_suffix("Cortex XSOAR")
 
     def add_user_to_departing_employee(self, username, departure_epoch=None, note=None):
-        de = self._sdk.employee_case_management.departing_employee
         try:
-            res = de.create_departing_employee(username, departure_epoch=departure_epoch, notes=note)
+            user_id = self.get_user_id(username)
+            self._sdk.detectionlists.departing_employee.add(
+                user_id, departure_epoch=departure_epoch
+            )
+            not note or self._sdk.detectionlists.update_user_notes(note)
         except Exception:
             return None
-        return res.json().get('caseId')
+        return user_id
 
     def fetch_alerts(self, start_time, event_severity_filter=None):
-        alert_filter = []
-        # Create alert filter
-        if event_severity_filter:
-            alert_filter.append(Severity.is_in(list(map(lambda x: x.upper(), event_severity_filter))))
-        alert_filter.append(AlertState.eq(AlertState.OPEN))
-        alert_filter.append(DateObserved.on_or_after(start_time))
-        alert_query = AlertQuery(self._sdk.user_context.get_current_tenant_id(), *alert_filter)
-        alert_query.sort_direction = "asc"
-        alerts = self._sdk.security.alerts
         try:
-            res = alerts.search_alerts(alert_query)
+            query = self._create_alert_query(event_severity_filter, start_time)
+            res = self._sdk.alerts.search_alerts(query)
         except Exception:
             return None
-        return res.json().get('alerts')
+        return res["alerts"]
+
+    def _create_alert_query(self, event_severity_filter, start_time):
+        alert_filters = []
+        # Create alert filter
+        if event_severity_filter:
+            f = event_severity_filter
+            severity_filter = (f.upper() if isinstance(f, str) else list(map(lambda x: x.upper(), f)))
```

Comment:
this appears to fix the bug re: single filters, nice

Reply:
actually wait, does it? this look like it might still just return a string if it's an instance of  a string? wouldn't we want `[f.upper()]` instead of just `f.upper()`?

Reply:
Yeah makes sense.. Sorry, I hadn't actually tested this yet. I am going to write a unit test for it too